setup.py

```diff
@@ -24,30 +24,50 @@ def get_tag(self):
         return python, abi, plat
 
 
+def abs_machine():
+    machine = platform.machine()
+
+    pointer_bits = struct.calcsize("P") * 8
+    if pointer_bits not in (32, 64):
+        raise Exception("Unsupported pointer size")
+
+    is_64 = pointer_bits == 64
+
+    # x86 based archs
+    if machine in ('AMD64', 'x86_64', 'i686-64', 'i386', 'i686', 'x86'):
+        return "x86_64" if is_64 else "i686"
+    # arm based archs
+    elif machine in ('aarch64', 'arm64', 'armv6l', 'armv7l'):
```

Comment:
fwiw, I'm not 100% certain if the binary will work on armv6l, since the QEMU emulation targets armv7. I think I have an armv6 raspberry pi 1 somewhere, can test on that.

Reply:
> fwiw, I'm not 100% certain if the binary will work on armv6l, since the QEMU emulation targets armv7.

sdist supposed to be buildable out of QEMU too.


> I think I have an armv6 raspberry pi 1 somewhere, can test on that.

Thanks you.

Reply:
```
pi@raspberrypi1:~/curl-impersonate $ ./curl-impersonate-chrome
Illegal instruction
```
I believe armv7 is a superset of the v6 instructions, so v6 will run on v7 but not guaranteed the other way around. If we're interested in v6 compatibility, I can investigate changing the container build pipeline to use v6.

Reply:
After a quick look, building curl-impersonate for armv6 via docker is going to be a massive headache since there are limited base images with armv6 support.  It would be easier to run the build on an armv6 raspberry pi, but I would wait and see if there's a need from the community before investing the time and equipment to produce armv6 wheels.

Reply:
Cross compile may also be an option...

Reply:
Is armv6 still relevant? I think it's only used for RPi 1 and very old android phones, right?